BrickBreaker.js

```diff
@@ -156,22 +161,41 @@ class Ball{
 		}
 
 		if (cross(this.x, this.y, new_x, new_y, leftBorder - this.radius, topBorder-this.radius, rightBorder + this.radius, topBorder - this.radius)){
-			this.horizontalCollision();
-			brick.collision();
+			if(this.isPower){
+				brick.collision();
+			}
+			else{
+				this.horizontalCollision();
+				brick.collision();
+			}
 		}
 		if (cross(this.x, this.y, new_x, new_y, leftBorder - this.radius, bottomBorder + this.radius, rightBorder + this.radius, bottomBorder + this.radius)){
-			this.horizontalCollision();
-			brick.collision();
+			if(this.isPower){
+				brick.collision();
+			}
+			else{
+				this.horizontalCollision();
+				brick.collision();
+			}
 		}
 		if (cross(this.x, this.y, new_x, new_y, leftBorder - this.radius, topBorder - this.radius, leftBorder - this.radius, bottomBorder + this.radius)){
-			this.verticalCollision();
-			brick.collision();
+			if(this.isPower){
+				brick.collision();
+			}
+			else{
+				this.verticalCollision();
+				brick.collision();
+			}
 		}
 		if (cross(this.x, this.y, new_x, new_y, rightBorder + this.radius, topBorder - this.radius, rightBorder + this.radius, bottomBorder + this.radius)){
-			this.verticalCollision();
-			brick.collision();
+			if(this.isPower){
+				brick.collision();
+			}
+			else{
+				this.verticalCollision();
+				brick.collision();
+			}
```

Comment:
공통적으로 적용되는 `brick.collision`은 if문 밖으로 빼면 더 간결할 것 같습니다!
```suggestion
                        brick.collision();
                        if(this.isPower) this.verticalCollision();
```